data/txHandler.go

```diff
@@ -302,14 +305,26 @@ func (handler *TxHandler) processIncomingTxn(rawmsg network.IncomingMessage) net
 			logging.Base().Warnf("Received a non-decodable txn: %v", err)
 			return network.OutgoingMessage{Action: network.Disconnect}
 		}
+		consumed = dec.Consumed()
 		ntx++
+		if ntx >= config.MaxTxGroupSize {
```

Comment:
current implementation accepts these large groups but they are rejected by txn pool.
the new implementation takes first 16 entries from large groups. This would not break consensus but some non upgraded proposers might not have such transactions when they receive a proposal.
@cce @algonautshant opinion?

Reply:
The problem with altering the incoming transaction group which has more than the max group size and letting it in is that you will have an impact on the consensus:
I can send a transaction group which exceeds the max group size, but set the TxGroupHashes only on the first MaxTxGroupSize transactions, and sign only on the hash of the first MaxTxGroupSize transactions. 

Before your changes, my transaction group will fail, but now will pass. You will be "correcting" my defective transaction and accepting it!

Reply:
I disagree. Protocol does not specify size of upcoming messages. It is impossible to "fork" if some nodes start receiving more valid tx groups. Although I exclude this change from this PR (it is still in dedup cache PR)

Reply:
I agree with you that it is impossible to fork this way.
But you will get for the same consensus version, one node accepting a block, while another rejecting. 
Anyway, I'm glad it's gone now :-)